proposals/dao/fip_55.ts

```diff
@@ -0,0 +1,62 @@
+import { ethers } from 'hardhat';
+import chai, { expect } from 'chai';
+import CBN from 'chai-bn';
+import {
+  DeployUpgradeFunc,
+  NamedContracts,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '../../types/types';
+import { PCVGuardian } from '@custom-types/contracts';
+
+chai.use(CBN(ethers.BigNumber));
+
+/*
+PCV Guardian
+
+DEPLOY ACTIONS:
+
+1. Deploy PCV Guardian
+
+DAO ACTIONS:
+1. Grant PCV Guardian GUARDIAN_ROLE
+2. Grant PCV Guardian PCV_CONTROLLER_ROLE
+
+*/
+
+export const deploy: DeployUpgradeFunc = async (deployAddress, addresses, logging = false) => {
+  const { core, feiDAOTimelock, compoundEthPCVDeposit, ethReserveStabilizer, aaveEthPCVDeposit } = addresses;
+
+  if (!core) {
+    throw new Error('An environment variable contract address is not set');
+  }
+
+  // 1. Deploy PCV Guardian
+  const factory = await ethers.getContractFactory('PCVGuardian');
+  const safeAddresses = [feiDAOTimelock, compoundEthPCVDeposit, ethReserveStabilizer, aaveEthPCVDeposit];
+  const pcvGuardian = await factory.deploy(core, safeAddresses);
+
+  await pcvGuardian.deployTransaction.wait();
+
+  logging && console.log('pcvGuardian: ', pcvGuardian.address);
+
+  return {
+    pcvGuardian
+  } as NamedContracts;
+};
+
+export const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('No setup');
+};
+
+export const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('No teardown');
+};
+
+export const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts) => {
+  const pcvGuardian: PCVGuardian = contracts.pcvGuardian as PCVGuardian;
```

Comment:
We shouldn't need to cast to type of PCVGuardian as the artifact that is loaded from contracts should be correct.